pkg/kv/kvserver/spanset/batch.go

```diff
@@ -975,10 +975,26 @@ func DisableReaderAssertions(reader storage.Reader) storage.Reader {
 	}
 }
 
+// DisableWriterAssertions unwraps any storage.Writer implementations that may
+// assert access against a given SpanSet.
+func DisableWriterAssertions(writer storage.Writer) storage.Writer {
+	switch v := writer.(type) {
+	case spanSetWriter:
+		return DisableWriterAssertions(v.w)
+	case *spanSetBatch:
+		return DisableWriterAssertions(v.spanSetWriter.w)
+	case *spanSetWriteBatch:
```

Comment:
nit: order cases from small to large (Writer, WriteBatch, Batch). A trick to not forget anything.